uni/lib/view/exams/widgets/exam_filter_form.dart

```diff
@@ -37,32 +37,35 @@ class ExamFilterFormState extends State<ExamFilterForm> {
       content: SizedBox(
           height: 230.0,
           width: 200.0,
-          child: getExamCheckboxes(widget.filteredExamsTypes, context)),
+          child: GetExamCheckboxes(widget.filteredExamsTypes, context)),
     );
   }
+}
 
-  Widget getExamCheckboxes(
-      Map<String, bool> filteredExams, BuildContext context) {
-    filteredExams.removeWhere((key, value) => !Exam.types.containsKey(key));
-    return ListView(
-        children: List.generate(filteredExams.length, (i) {
-      final String key = filteredExams.keys.elementAt(i);
-      if (!Exam.types.containsKey(key)) return const Text("");
-      return CheckboxListTile(
-          contentPadding: const EdgeInsets.all(0),
-          title: Text(
-            key,
-            overflow: TextOverflow.ellipsis,
-            softWrap: false,
-            maxLines: 2,
-          ),
-          key: Key('ExamCheck$key'),
-          value: filteredExams[key],
-          onChanged: (value) {
-            setState(() {
-              filteredExams[key] = value!;
-            });
-          });
-    }));
-  }
+class GetExamCheckboxes extends StatelessWidget {
```

Comment:
Widget names should not be an action, but rather the description of the view. In this case, for example: `FilteredExamsList`. Also, widgets do not need to receive an explicit context, since a new node is created on the tree, hence a new context.
Also, have in mind that it makes sense to extract widgets only if one of these apply:

- They have state, thus possibly reducing parent rebuilding
- They are reusable